tests/sentry/notifications/utils/test_get_most_specific.py

```diff
@@ -0,0 +1,104 @@
+from unittest import TestCase
+
+from sentry.models import User
+from sentry.notifications.helpers import (
+    get_highest_notification_setting_value,
+    get_most_specific_notification_setting_value,
+)
+from sentry.notifications.types import (
+    NotificationScopeType,
+    NotificationSettingOptionValues,
+    NotificationSettingTypes,
+)
+from sentry.types.integrations import ExternalProviders
+
+
+class GetMostSpecificNotificationSettingValueTestCase(TestCase):
+    def setUp(self) -> None:
+        self.user = User(id=1)
+
+    def test_get_most_specific_notification_setting_value_empty_workflow(self):
+        value = get_most_specific_notification_setting_value(
+            notification_settings_by_scope={},
+            user=self.user,
+            parent_id=1,
+            type=NotificationSettingTypes.WORKFLOW,
+        )
+        assert value == NotificationSettingOptionValues.SUBSCRIBE_ONLY
```

Comment:
does this mean that if the user has no setting for workflow notifications, we default to subscribe only?

Reply:
Yeah a brand new user would have no notification settings and would be opted into workflow emails.
```python
# This mapping represents how to interpret the absence of a DB row for a given
# provider. For example, a user with no NotificationSettings should be opted
# into receiving emails but no Slack messages.
NOTIFICATION_SETTING_DEFAULTS = {
    ExternalProviders.EMAIL: {
        NotificationSettingTypes.DEPLOY: NotificationSettingOptionValues.COMMITTED_ONLY,
        NotificationSettingTypes.ISSUE_ALERTS: NotificationSettingOptionValues.ALWAYS,
        NotificationSettingTypes.WORKFLOW: NotificationSettingOptionValues.SUBSCRIBE_ONLY,
    },
    ExternalProviders.SLACK: {
        NotificationSettingTypes.DEPLOY: NotificationSettingOptionValues.NEVER,
        NotificationSettingTypes.ISSUE_ALERTS: NotificationSettingOptionValues.NEVER,
        NotificationSettingTypes.WORKFLOW: NotificationSettingOptionValues.NEVER,
    },
}
```